sdk/src/azure/core/az_log.c

```diff
@@ -57,8 +79,14 @@ static bool _az_log_write_engine(bool log_it, az_log_classification classificati
                            // is going to succeed and return from the function.
   }
 
-  for (az_log_classification const* cls = classifications; *cls != AZ_LOG_END_OF_LIST; ++cls)
+  az_log_classification const* cls = classifications;
+  while (*cls != AZ_LOG_END_OF_LIST)
```

Comment:
Why did you remove the fol loop? If=t was line instead of 3, all iterations and exit logic in 1 place.

Reply:
Good point, will revert.